concent_api/core/payments/bankster.py

```diff
@@ -147,3 +157,130 @@ def finalize_payment(
+    requestor_deposit_value = service.get_deposit_value(client_eth_address=requestor_ethereum_address)
+
+    # Concent defines time T0 equal to oldest payment_ts from passed SubtaskResultAccepted messages from
+    # subtask_results_accepted_list.
+    oldest_payments_ts = min(subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances)
+
+    current_time = get_current_utc_timestamp()
+
+    # Concent gets list of transactions from payment API where timestamp >= T0.
+    list_of_transactions = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
+        requestor_eth_address=requestor_ethereum_address,
+        provider_eth_address=provider_ethereum_address,
+        payment_ts=oldest_payments_ts,
+        current_time=current_time,
+        transaction_type=TransactionType.BATCH,
+    )
+
+    # Concent defines time T1 equal to youngest timestamp from list of transactions.
+    if len(list_of_transactions) == 0:
+        t1_is_bigger_than_payments_ts = None
+    else:
+        youngest_transaction = max(transaction.closure_time for transaction in list_of_transactions)
+        # Concent checks if all passed SubtaskResultAccepted messages from subtask_results_accepted_list
+        # have payment_ts < T1
+        t1_is_bigger_than_payments_ts = any(
+            youngest_transaction > subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances
+        )
+
+    # Any of the items from list of overdue acceptances
+    # matches condition current_time < payment_ts + PAYMENT_DUE_TIME
+    acceptance_time_overdue = any(
+        current_time < subtask_results_accepted.payment_ts + settings.PAYMENT_DUE_TIME
+        for subtask_results_accepted in acceptances
+    )
+
+    if t1_is_bigger_than_payments_ts or acceptance_time_overdue:
+        return ClaimPaymentInfo(0, 0, timestamp_error=True)
+
+    # Concent gets list of forced payments from payment API where T0 <= payment_ts + PAYMENT_DUE_TIME.
+    list_of_forced_payments = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
+        requestor_eth_address=requestor_ethereum_address,
+        provider_eth_address=provider_ethereum_address,
+        payment_ts=oldest_payments_ts + settings.PAYMENT_DUE_TIME,  # Im not sure, check it please
+        current_time=current_time,
+        transaction_type=TransactionType.FORCE,
+    )
+
+    (amount_paid, amount_pending) = sum_amount_price_for_provider(
+        list_of_forced_payments=list_of_forced_payments,
+        list_of_payments=list_of_transactions,
+        subtask_results_accepted_list=acceptances,
+    )
+
+    if amount_pending == 0:
+        requestors_claim_payment_info = ClaimPaymentInfo(0, 0)
+    elif requestor_deposit_value == 0:
+        requestors_claim_payment_info = ClaimPaymentInfo(0, amount_pending)
+    else:
+        request_can_pay = min(amount_pending, requestor_deposit_value)
```

Comment:
`requestor_payable_amount`? It's name right now suggest that it's `bool` whereas it is in fact an `int`.

Reply:
:+1: done